front/pages/api/w/[wId]/groups/[gId]/index.ts

```diff
@@ -0,0 +1,133 @@
+import type { GroupType, UserType, WithAPIErrorResponse } from "@dust-tt/types";
+import { PatchGroupRequestBodySchema } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { apiError } from "@app/logger/withlogging";
+
+export type GetGroupResponseBody = {
+  group: GroupType & {
+    members: UserType[];
+  };
+};
+
+export type PatchGroupResponseBody = {
+  group: GroupType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<GetGroupResponseBody | PatchGroupResponseBody>
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const { gId } = req.query;
+
+  if (typeof gId !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "group_not_found",
+        message: "The group you requested was not found.",
+      },
+    });
+  }
+
+  const group = await GroupResource.fetchById(auth, gId);
+  // Check if the user has access to the group to get members list
+  if (!group) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "group_not_found",
+        message: "The group you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const members = await group.getActiveMembers(auth);
+      if (
```

Comment:
This yields an interesting question of how we want to standardize our authorization code.

Looking for member in a group seems at odd with checking that the group is in the auth group list which is the standard way?

Do we have standardized generic authy canReady/canWrite type of functions? Or do we generally put these on resources?

In both cases we should likely use this here instead of checking members which is a bit brittle and will be hard to find in the future if we want to audit/check/change all of our authorization code?

Reply:
Actually in that case yes we could check that the group is in the list of authenticator groups (the list of groups of the users). The members were already retrieved here to be returned so one way or the other is fine (although i could move the `await group.getActiveMembers(auth)` inside the test)

We have canReady/canWrite , that works on acl , but we don't have acls on groups themselves for now. We could perfectly add add an acl on groups, but we'll need to identify an "admin" group of all users who have isAdmin on the workspace, to know who can write. But using canRead() should be ok.

Reply:
Interesting question. I believe permissions should be managed by each resource. Often, a 404 error is returned when a user lacks access, preventing resource discovery. This aligns with returning null from a resource method when the user doesn't have access.

I'm a bit on the fence to have canWrite/canRead options as we will end up with a lot of duplicated logic spread through the codebase where as we could keep permissions logic tight to the resource/auth.